### PR TITLE
change format

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
@alena1108 Could you review this again? Changed `ipc_mode` to `ipc`and not exporting null fields for @joshwget.

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/6404

File: code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/service/impl/ServiceDiscoveryApiServiceImpl.java [hunk 1]
```diff
@@ -363,11 +363,21 @@ private void populateBlkioOptions(Map<String, Object> cattleServiceData, Map<Str
                         }
                     }
                 }
-                composeServiceData.put("blkio_weight_device", deviceWeight);
-                composeServiceData.put("device_read_bps", deviceReadBps);
-                composeServiceData.put("device_read_iops", deviceReadIops);
-                composeServiceData.put("device_write_bps", deviceWriteBps);
-                composeServiceData.put("device_write_iops", deviceWriteIops);
+                if (deviceWeight.size() != 0) {
```
Comment (alena1108) [inline code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/service/impl/ServiceDiscoveryApiServiceImpl.java]: Nit: you can use !deviceWeight.isEmpty() instead

File: code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/util/ServiceDiscoveryUtil.java [hunk 2]
```diff
@@ -157,6 +157,12 @@ public static String getServiceSuffixFromInstanceName(String instanceName) {
             launchConfigData = DataAccessor.fields(service)
                     .withKey(ServiceConstants.FIELD_LAUNCH_CONFIG).withDefault(Collections.EMPTY_MAP)
                     .as(Map.class);
+            // if the value is empty, do not export
+            for (String key: launchConfigData.keySet()) {
```
Comment (alena1108) [inline code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/util/ServiceDiscoveryUtil.java]: It is not really safe to remove key from map while iterating over it. You can introduce some temporary value for iteration (and addAll items from the original launchConfigData); and then remove form launchConfigData should work

Comment (StrongMonkey) [inline code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/util/ServiceDiscoveryUtil.java]: yeah it is true. i ll make a change.

**Comment by StrongMonkey** [comment]
@alena1108  updated.

File: code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/util/ServiceDiscoveryUtil.java (hunk 2 above)
Comment (ibuildthecloud) [inline code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/util/ServiceDiscoveryUtil.java]: FYI this works in go.  We should probably just rewrite this all in go 😝

Comment (joshwget) [inline code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/util/ServiceDiscoveryUtil.java]: Time for another microservice!

**Comment by alena1108** [comment]
LGTM

**Pull request merged**